teuthology/test/task/test_pcp.py

```diff
@@ -360,3 +361,18 @@ def test_end(self, m_grafana, m_graphite, m_makedirs):
         # end() should have called write_html() a second time by now
         assert len(task.graphite.write_html.call_args_list) == 2
         assert isinstance(task.stop_time, int)
+
+    @patch('os.makedirs')
+    @patch('teuthology.task.pcp.GrafanaGrapher')
+    @patch('teuthology.task.pcp.GraphiteGrapher')
+    def test_end_16049(self, m_grafana, m_graphite, m_makedirs):
+        # http://tracker.ceph.com/issues/16049
+        # Jobs were failing if graph downloading failed. We don't want that.
+        self.ctx.archive = '/fake/path'
+        with self.klass(self.ctx, self.task_config) as task:
+            task.graphite.download_graphs.side_effect = \
+                requests.ConnectionError
+        # Even though downloading graphs failed, we should have called
+        # write_html() a second time
+        assert len(task.graphite.write_html.call_args_list) == 2
```

Comment:
I realize there's always overkill, but, given that it's easy to check call args, would you want to do that here (i.e. static the first time, no static the second)?  That could subsume the "number of calls" check